rllib/agents/sac/sac_policy.py

```diff
@@ -155,6 +158,10 @@ def actor_critic_loss(policy, model, _, train_batch):
 
     # Q-values for current policy (no noise) in given current state
     q_t_det_policy = model.get_q_values(model_out_t, policy_t)
+    if policy.config["twin_q"]:
+        twin_q_t_det_policy = model.get_q_values(model_out_t, policy_t)
```

Comment:
Should this be using `model.get_twin_q_values`?  It seems the `min` on the next line would just take the minimum of identical values.

Reply:
You are right... could you see how changing this line of code does for `Pendulum-v0` in tuned examples? Hopefully it should improve sample efficiency.

Reply:
FYI @sven1977

Reply:
![image](https://user-images.githubusercontent.com/37295291/75244515-d0b54000-57c3-11ea-9573-cf6d56552489.png)
blue = with fix, orange = without fix.  It doesn't seem to have made a huge difference, although perhaps we wouldn't expect it to on such a small task?  I would give HalfCheetah-v3 a go, however I do not have a MuJuCo license.

Reply:
Send me a message at michael.luo@berkeley.edu. I'll be happy to help you :).

Reply:
I think `Pendulum-v0` is probably too trivial environment for the min-Q trick to actually have an effect. If you compare this in environment like `Ant-v3` or `Humanoid-v3`, I'd expect both the final performance and also the learning speed to be much better with min-Q.

Reply:
![image](https://user-images.githubusercontent.com/37295291/75423359-7dafca00-5936-11ea-8ff3-bd77be832da9.png)
Here are the results for HalfCheetah-v3 using halfcheetah-sac.yaml from the repository.  I checked out the 0.8.2 tag as master had stability issues (`raise MujocoException(warn + 'Check for NaN in simulation.')`), and commented out the `exploration_enabled` line in the YAML file as I was getting `Unknown config parameter `exploration_enabled`` errors.

The grey line is WITH the fix, green line WITHOUT.  So a bit of a counter intuitive result.  Could it be that because it was hyper parameter tuned without the fix, changing the algorithm in this way has in fact caused degradation?  If so we'd want to re-tune with the fix to do a like for like comparison (unfortunately I don't have resources for this at the moment).

Reply:
`raise MujocoException(warn + 'Check for NaN in simulation.'` in case of SAC is very likely due to numerically unstable implementation of the squashed head (`SquashedGaussian` here if I remember seeing correctly) of the policy. See e.g. https://github.com/tensorflow/probability/issues/318.

Reply:
Ignore what I said about `SquashedGaussian`, since this PR is still using the `SquashBijector`. That should be okay.

Reply:
Actually, I was using `SquashedGaussian` (`master` latest)  when I saw the NaN error, and it disappeared when I downgraded.  I commented on this PR because it was where the `get_q_values` line in question was introduced.

Reply:
@sven1977

Reply:
Btw @mattearllongshot could you report your TF/TFP version and how to reproduce? The reason is we didn't see the instability in our testing.

Reply:
@mattearllongshot I'll take another look at the SquashedGaussian code to see, whether I can break it as well. Haven't seen any issues when running it against HalfCheetah a couple of times last WE (actually performed slightly better than our previous baseline (~14k vs 13k)).

Reply:
@mattearllongshot And thanks for catching that bug. :)

Reply:
Here's a related tensorflow-probability issue if that helps: https://github.com/tensorflow/probability/issues/318.

Reply:
I'm using

```
tensorflow==2.1.0
tensorflow-estimator==2.1.0
tensorflow-probability==0.9.0
```

on commit 9964657815fdd125d364ef8a09bd3cd046399611, running `rllib train -f ./rllib/tuned_examples/halfcheetah-sac.yaml`.  On this occasion I saw the `Check for NaN in simulation.` error after 32 iterations

Reply:
Shall I raise a bug for these two so they don't get lost?

Reply:
or rather, two bugs